File: src/test/test_type/2.fun_pro/c_code/test1.c
#include <stdio.h>

float a,sum;

void add(int x, int y)
{
	int result;
	result = x+y;
	printf("%d", result);
}

int main()
{
	a = 3.5;
	add(a, 4);
	return 0;
}

File: test/test_type/2.fun_pro/test1.pas
program test(input,output);
var a, sum: real;
procedure add(x, y: integer);
var result: integer;
begin
  result:=x+y;
  write(result);
end;

begin
a:=3.5;
add(a, 4); // 类型不匹配错误，传递实数参数给接受整数参数的函数
end.